src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/FileResultExecutorBase.cs

```diff
@@ -200,6 +204,11 @@ private static void SetAcceptRangeHeader(HttpResponse response)
                     etag: etag,
                     matchFoundState: PreconditionState.ShouldProcess,
                     matchNotFoundState: PreconditionState.PreconditionFailed);
+
+                if (ifMatchState == PreconditionState.PreconditionFailed)
+                {
+                    logger.IfMatchPreconditionFailed(etag);
+                }
```

Comment:
There should be logging for the case where IfNoneMatchState is PreconditionFailed.

Instead, I think there should be logging within the GetEtagMatchState method or GetMaxPreconditionState method that logs the state instead of only logging precondition failed. Maybe the other states can be info level and the failure can be a debug/error - not sure which one is better for this case.

Reply:
> There should be logging for the case where IfNoneMatchState is PreconditionFailed.

It doesn't have this state

> Instead, I think there should be logging within the GetEtagMatchState method or GetMaxPreconditionState method that logs the state instead of only logging precondition failed. Maybe the other states can be info level and the failure can be a debug/error - not sure which one is better for this case.

We do not want Info level for these cases as its too much into detail and ideally is Debug. Error also doesn't seem appropriate as its a pre-condition and failure is expected.

Reply:
Okay